Replacing GitVersion 5 with 6

raandree opened the pull request:
# Pull Request

## Pull Request (PR) description

**The is a breaking change** that replaces the GitVersion 5 support with GitVersion 6. The schema for [6.3](https://github.com/GitTools/GitVersion/blob/main/schemas/6.3/GitVersion.configuration.json) is not compatible with [5.12](https://github.com/GitTools/GitVersion/blob/main/schemas/5.12/GitVersion.configuration.json).

GitVersion 6 does no longer support zero-padded versions like `3.0.1-preview0001`, instead it looks like `3.0.1-preview.1` (FullSemVer). To be compatible with NuGet 2.0 (PowerShell, Chocolatey), the build task `GitVersion` does the padding for pre-releases.

References:
- [Versioning Software](https://gist.github.com/Jaykul/5687dc25b081e61e41de2a5635b506aa#file-about-versioning-md)
- [[Bug] NugetPreReleaseTag does not respect padding setting #2632](https://github.com/GitTools/GitVersion/issues/2632)

### Added

- Added support for GitVersion 6 (Breaking Change).
  - Updated `GitVersion.yml` config file.
  - Removed the GitVersion task in Azure DevOps pipeline.
  - Added a task to the Azure DevOps pipeline to Install GitVersion.
  - Added a new build task to Sampler: `GitVersion`.
  - Updated `Get-SamplerBuildVersion` to be compatible with GitVersion 6.
  - Updated `Get-SamplerBuildVersion` Pester tests.
  - Changed `README.md`.

### Changed

- Set GitVersion mode to [`Manual Deployment` ](https://gitversion.net/docs/reference/modes/manual-deployment).
  > "Having not the necessity to deploy the build artifacts on every commit is an indication of using the Manual Deployment mode. This mode can be used to remain on the same pre-released version until it has been deployed dedicatedly.
- Updated `GitVersion.yml` config file to work with GitVersion 6.

## Task list

- [x] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [x] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [x] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [x] Documentation added/updated in README.md.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gaelcolas/Sampler/529)
<!-- Reviewable:end -->

raandree committed:
Considering the environment variable 'ModuleVersion' if it exists

raandree committed:
Considering the environment variable 'ModuleVersion' if it exists

raandree committed:
Added support for GitVersion which is a breaking change

raandree committed:
Added support for GitVersion which is a breaking change

raandree committed:
Updated launch configs

raandree committed:
Updated launch configs

raandree committed:
Adapting tests to work with GitVersion 6

raandree committed:
Adapting tests to work with GitVersion 6

raandree committed:
Updated changelog

raandree committed:
Updated changelog

raandree committed:
Change deployment mode from ContinuousDelivery to ManualDeployment

raandree committed:
Change deployment mode from ContinuousDelivery to ManualDeployment

raandree committed:
Updated changelog

raandree committed:
Fixed typo

raandree committed:
Fixed typo

gaelcolas reviewed (changes requested):
Do we _have_ to make it a breaking change?
If the template is not the same that's fine, but I feel existing repos can continue to work with gitversion v5.
There's another couple of points I'd like to discuss to clarify, but generally I think the way it works is ok.
Let's discuss...

raandree committed:
Merge branch 'main' into feature/gitversion6

raandree committed:
Changes according to the code review

raandree committed:
Changes according to the code review

raandree committed:
Added test to check for environment variable

raandree committed:
Added test to check for environment variable

raandree committed:
Merge branch 'main' into feature/gitversion6

raandree committed:
Merge branch 'main' into feature/gitversion6

johlju committed:
Fix Script Analyzer QA tests failing in various scripts (#540)

Mynster9361 committed:
Add mac setup instructions to README (#532)

Ligtorn committed:
Fixed issue with changelog check (#541)

gaelcolas committed:
fixing http.sslbackend schannel command because it failed

gaelcolas committed:
updating changelog

gaelcolas committed:
Updating ChangeLog since v0.119.0 +semver:skip (#548)

gaelcolas committed:
fixing http.sslbackend schannel command because it failed

dan-hughes committed:
Fix Gallery Preview Badge

dan-hughes committed:
Update PublishCodeCoverage to v2 (#552)

gaelcolas committed:
New pack task using PSResourceGet (#555)

DennisL68 committed:
Moved large parts of the ReadMe to the Wiki (#488)

ahpooch committed:
Updated comment based help for Resolve-Dependency.ps1 (#537)

raandree committed:
Merge branch 'feature/gitversion6' of https://github.com/raandree/Sampler into feature/gitversion6

johlju commented:
/azp run

raandree committed:
fix: correct GitVersion 6 config for version bump regexes and feature branch label

The version-bump-message regexes were missing the '+semver:' prefix,
causing any commit containing words like 'breaking' or 'feature' to
trigger unintended Major/Minor bumps (e.g. 0.114.0 -> 1.0.0).

The feature branch regex lacked the (?<BranchName>.+) named capture
group, so the '{BranchName}' label token was never resolved.

Changes:
- Add '\+semver:' prefix to major/minor/patch-version-bump-message
- Fix feature branch regex to '^features?[\/-](?<BranchName>.+)'

raandree committed:
```
fix: pin Plaster to 1.1.4 to avoid breaking change in 2.0.0

Plaster 2.0.0 (published 2026-04-13) removed the internal function
'GetPlasterManifestPathForCulture', which breaks Add-Sample's dynamic
parameter resolution and causes the 'SourceDirectory' parameter to
not be found during tests.
```

raandree committed:
test: add stubs for gitversion in environment variable tests

raandree commented:
@gaelcolas, I have rebased the branch. Can you have a look at it, please?

johlju commented:
/azp run

gaelcolas commented:
FTR, I'll discuss with @raandree what to do as I'd rather avoid a breaking change, but updating a task would be good, or alternatively have the task pick up the version and decide on the behaviour.

raandree committed:
fix: standardize formatting of RequiredModules.psd1 entries

raandree committed:
fix(gitversion): skip AzDo pipeline variable writes when not running on an AzDo agent

Guard the ##vso[task.setvariable] and ##vso[build.updatebuildnumber] writes with a check for $env:TF_BUILD so local builds don't emit noisy Azure DevOps logging commands.
